Source/OCMockTests/OCMArgTests.m

```diff
@@ -101,4 +101,10 @@ - (void)testHandlesNonObjectPointersGracefully
     XCTAssertEqual([OCMArg resolveSpecialValues:nonObjectPointerValue], nonObjectPointerValue, @"Should have returned value as is.");
 }
 
+- (void)testThrowsForNilArgumentToUnretainedObjects
+{
+    XCTAssertThrows([OCMArg unretainedObject:nil]);
```

Comment:
This should verify that the message is correct. You may wish to invent a macro like this one to make that easier:

```objc
#define XCTAssertThrowsEquivalent(EXPECTED, THROWING_CALL) \
({ \
    BOOL threw = NO;\
    NSException *capturedException = nil; \
    @try { \
        THROWING_CALL; \
    } @catch (NSException *thrownException) { \
        capturedException = [thrownException retain]; \
        threw = YES; \
    } @finally { \
        XCTAssertTrue(threw, @"Expected exception was not thrown: %@", EXPECTED); \
        if (threw) {\
            if ([[capturedException name] isEqualToString:@"OCMStubbedException"]) { \
                id actualException = [[[capturedException userInfo] objectForKey:@"exception"] retain]; \
                XCTAssertNotNil(actualException, @"Exception thrown by OCMock doesn't include the expected underlying exception: %@\n%@", EXPECTED, capturedException); \
                XCTAssertExceptionMatchesException(EXPECTED, actualException); \
                [actualException release]; actualException = nil; \
            } else { \
                XCTAssertExceptionMatchesException(EXPECTED, capturedException); \
            } \
        } \
        [capturedException release]; capturedException = nil; \
    } \
})
```

That depends on a plethora of other macros:

```objc
#define XCTAssertExceptionMatchesException(EXPECTED, ACTUAL) \
({ \
    XCTAssertStringMatchesString([EXPECTED name], [ACTUAL name]); \
    XCTAssertStringMatchesString([(NSException *)EXPECTED reason], [(NSException *)ACTUAL reason]); \
    XCTAssertDictionaryMatchesDictionary([EXPECTED userInfo], [ACTUAL userInfo]); \
})

#define XCTAssertStringMatchesString(EXPECTED, ACTUAL) \
({ \
    if (EXPECTED == nil) { \
        if (ACTUAL != nil) { \
            XCTFail(@"String mismatch:\n%@\n%@", [XCTestingAsserts trimExcessiveValuesForLog:EXPECTED], [XCTestingAsserts trimExcessiveValuesForLog:ACTUAL]); \
        } \
    } else if (ACTUAL == nil) { \
        XCTFail(@"String mismatch:\n%@\n%@", [XCTestingAsserts trimExcessiveValuesForLog:EXPECTED], [XCTestingAsserts trimExcessiveValuesForLog:ACTUAL]); \
    } else { \
        XCTAssertTrue([EXPECTED isEqualToString:ACTUAL], @"String mismatch:\n%@\n%@", [XCTestingAsserts trimExcessiveValuesForLog:EXPECTED], [XCTestingAsserts trimExcessiveValuesForLog:ACTUAL]); \
    } \
})

#define XCTAssertDictionaryMatchesDictionary(EXPECTED, ACTUAL) \
({ \
    if (EXPECTED != ACTUAL) { \
        if ((EXPECTED == nil) || (ACTUAL == nil)) { \
            XCTFail(@"Dictionary mismatch: %@ / %@", EXPECTED, ACTUAL); \
        } else { \
            XCTAssertTrue([EXPECTED isEqualToDictionary:ACTUAL], @"Dictionary mismatch: %@ / %@", EXPECTED, ACTUAL); \
        } \
    } \
})
```

Sorry, I know there's a lot here, but it really does make writing complete tests easier.

Reply:
I'm guessing that this should probably come in a separate PR and get applied to all of the throw checks. Thoughts Erik?

Reply:
Yes, smaller separate PRs are appreciated. ;-)